iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/utils/command/MultipleUnorderedOptionsPredicate.java

```diff
@@ -0,0 +1,169 @@
+import java.util.Set;
+import java.util.function.Predicate;
+import org.sonar.iac.docker.symbols.ArgumentResolution;
+
+import static org.sonar.iac.docker.checks.utils.command.CommandPredicate.Type.MATCH;
+import static org.sonar.iac.docker.checks.utils.command.CommandPredicate.Type.ZERO_OR_MORE;
+import static org.sonar.iac.docker.checks.utils.command.PredicateContext.Status.ABORT;
+import static org.sonar.iac.docker.checks.utils.command.PredicateContext.Status.FOUND_NO_PREDICATE_MATCH;
+
+public class MultipleUnorderedOptionsPredicate implements CommandPredicate {
+
+  final List<OptionPredicate> options;
+
+  final boolean shouldSupportAnyMatch;
+
+  public MultipleUnorderedOptionsPredicate(List<OptionPredicate> options) {
+    this.options = options;
+    this.shouldSupportAnyMatch = true;
+  }
+
+  public boolean isShouldSupportAnyMatch() {
+    return shouldSupportAnyMatch;
+  }
+
+  public OptionPredicate calculateAnyOptionMatchingExceptExpected() {
+    Predicate<String> noFlagFromExpectedOptions = options.stream()
+      .map(option -> option.flagPredicate.predicate.negate())
+      .reduce(s -> s.startsWith("-"), Predicate::and);
+    SingularPredicate anyFlagPredicate = new SingularPredicate(noFlagFromExpectedOptions, ZERO_OR_MORE);
+    // doesn't match on flags, cause flags start with '-'
+    SingularPredicate anyValuePredicate = new SingularPredicate(s -> !s.startsWith("-") && !s.startsWith("&&"), ZERO_OR_MORE);
+    return new OptionPredicate(anyFlagPredicate, anyValuePredicate);
+  }
+
+  /**
+   * true if any of the option predicates match
+   */
+  @Override
+  public boolean hasType(Type... types) {
+    for (OptionPredicate optionPredicate : options) {
+      if (optionPredicate.hasType(types)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  /**
+   * Used to match this predicate, by trying to match the contained list of {@link OptionPredicate options}.
+   * The list <code>options</code> can be matched in any order, and depending on <code>shouldSupportAnyMatch</code> with additional flags / 
+   * options that aren't provided as sensitive options.
+   */
+  @Override
+  public void match(PredicateContext context) {
+    // used to count how many options we already matched
+    Set<OptionPredicate> fulfilledOptions = new HashSet<>();
+
+    // set of options that aren't matched yet
+    List<OptionPredicate> workingSet = new ArrayList<>(this.options);
+
+    long expectedMatches = workingSet.stream().filter(optionPredicate -> optionPredicate.hasType(MATCH)).count();
+
+    OptionPredicate anyOptionExceptExpectedPredicate = this.calculateAnyOptionMatchingExceptExpected();
+
+    // initial initialization to step into while, could be solved with do-while
+    boolean anythingMatched = true;
+
+    while (anythingMatched && !workingSet.isEmpty()) {
+
+      if (context.areNoArgumentsToHandle()) {
+        if (!context.remainingPredicatesAreOptional()) {
+          context.setStatus(ABORT);
```

Comment:
The code should have more test : if I comment this line, I don't get any failed test.